services/subscriber/service.go

```diff
@@ -100,9 +100,10 @@ func (s *Service) waitForMetaUpdates() {
 		} else {
 			//Check that we haven't been closed before performing update.
 			s.mu.Lock()
-			if !s.closed {
+			if s.closed {
```

Comment:
This makes sense, though I think `closed` should also be set to `true` in `New Service`.